src/proposals/20240403/SparkEthereum_20240403.t.sol

```diff
@@ -0,0 +1,142 @@
+
+    int256 public latestAnswer;
+
+    constructor(int256 _latestAnswer) {
+        latestAnswer = _latestAnswer;
+    }
+    
+}
+
+contract SparkEthereum_20240403Test is SparkEthereumTestBase {
+
+    using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+
+    address internal constant KILL_SWITCH_ORACLE = 0x909A86f78e1cdEd68F9c2Fe2c9CD922c401abe82;
+
+    address internal constant WBTC_BTC_ORACLE  = 0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23;
+    address internal constant STETH_ETH_ORACLE = 0x86392dC19c0b719886221c78AB11eb8Cf5c52812;
+
+    constructor() {
+        id = '20240403';
+    }
+
+    function setUp() public {
+        vm.createSelectFork(getChain('mainnet').rpcUrl, 19523087);  // March 27, 2024
+        payload = deployPayload();
+
+        loadPoolContext(poolAddressesProviderRegistry.getAddressesProvidersList()[0]);
+    }
+
+    function testKillSwitchActivation() public {
+        IKillSwitchOracle kso = IKillSwitchOracle(KILL_SWITCH_ORACLE);
+
+        assertEq(aclManager.isRiskAdmin(KILL_SWITCH_ORACLE), false);
+        assertEq(kso.numOracles(), 0);
+        assertEq(kso.oracleThresholds(WBTC_BTC_ORACLE),  0);
+        assertEq(kso.oracleThresholds(STETH_ETH_ORACLE), 0);
+
+        GovHelpers.executePayload(vm, payload, executor);
+
+        assertEq(aclManager.isRiskAdmin(KILL_SWITCH_ORACLE), true);
+        assertEq(kso.numOracles(), 2);
+        assertEq(kso.oracleThresholds(WBTC_BTC_ORACLE),  0.95e8);
+        assertEq(kso.oracleThresholds(STETH_ETH_ORACLE), 0.95e18);
+        
+        // Sanity check the latest answers
+        assertEq(IChainlinkAggregator(WBTC_BTC_ORACLE).latestAnswer(),  0.99897716e8);
+        assertEq(IChainlinkAggregator(STETH_ETH_ORACLE).latestAnswer(), 0.999478607275791200e18);
+
+        // Should not be able to trigger either
+        assertEq(kso.triggered(), false);
+
+        vm.expectRevert("KillSwitchOracle/price-above-threshold");
+        kso.trigger(WBTC_BTC_ORACLE);
+
+        vm.expectRevert("KillSwitchOracle/price-above-threshold");
+        kso.trigger(STETH_ETH_ORACLE);
+
+        // Replace the aggregator with a mock
+        MockAggregator wbtcAggregator = new MockAggregator(0.95e8);
+        vm.store(
+            WBTC_BTC_ORACLE,
+            bytes32(uint256(2)),
+            bytes32((uint256(uint160(address(wbtcAggregator))) << 16) | 1)
+        );
+
+        assertEq(IChainlinkAggregator(WBTC_BTC_ORACLE).latestAnswer(),  0.95e8);
+
+        assertEq(_getBorrowEnabled(DAI),    true);
+        assertEq(_getBorrowEnabled(SDAI),   false);
+        assertEq(_getBorrowEnabled(USDC),   true);
+        assertEq(_getBorrowEnabled(WETH),   true);
+        assertEq(_getBorrowEnabled(WSTETH), true);
+        assertEq(_getBorrowEnabled(WBTC),   true);
+        assertEq(_getBorrowEnabled(GNO),    false);
+        assertEq(_getBorrowEnabled(RETH),   true);
+        assertEq(_getBorrowEnabled(USDT),   true);
+
+        kso.trigger(WBTC_BTC_ORACLE);
```

Comment:
Revert state and do the same for eth oracle

Reply:
done